### PR TITLE
Makes TRAIT_NOCLONELOSS mobs immune to clone damage

### LINKED ISSUE #45378: TRAIT_NOMETABOLISM species can't heal cloneloss with meds
#44846 didn't account for cloneloss, which isn't that uncommon due to cloning, slimes, etc. The only way for them to even heal it is a slime extract, asklepius' rod or a slime cookie.

Since those three species are very rare, I think it would be a decent solution to make them immune to clone damage as well.

### PR DESCRIPTION
**Pull request by spookydonut**
Revisiting this: #45471
Fixes #45378

## Changelog
:cl:
fix: TRAIT_NOMETABOLISM species are now immune to clone damage
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
For clarity, are things with this trait supposed to retain clone damage they had before the fact?

**Comment by spookydonut** [comment]
> 
> 
> For clarity, are things with this trait supposed to retain clone damage they had before the fact?

can you make things trigger on trait gain?

**Comment by ExcessiveUseOfCobblestone** [comment]
long answer: yes, wyci
short answer: no

**Comment by PKPenguin321** [comment]
>can you make things trigger on trait gain?

You either put a check for the trait on life() or some other relevant proc that clears damage when reached (not advisable), or track down every instance of the trait being added and manually remove clone damage at each instance

**Comment by spookydonut** [comment]
I could just make it not count clonedamage for the purposes of dying etc

File: code/modules/mob/living/damage_procs.dm [hunk 1]
```diff
@@ -207,15 +207,15 @@
 	return cloneloss
 
 /mob/living/proc/adjustCloneLoss(amount, updating_health = TRUE, forced = FALSE)
-	if(!forced && (status_flags & GODMODE))
+	if(!forced && (status_flags & GODMODE) || HAS_TRAIT(src, TRAIT_NOCLONELOSS))
```
Comment (AnturK) [inline code/modules/mob/living/damage_procs.dm]: This still should happen when forced?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/damage_procs.dm (hunk 1 above)
Comment (spookydonut) [reply code/modules/mob/living/damage_procs.dm]: Oh true, i need to add brackets

Comment (spookydonut) [inline code/modules/mob/living/damage_procs.dm]:
```suggestion
	if(!forced && ( (status_flags & GODMODE) || HAS_TRAIT(src, TRAIT_NOCLONELOSS)) )
```

File: code/modules/mob/living/damage_procs.dm [hunk 2]
```diff
@@ -207,15 +207,15 @@
...
+	if(!forced && ( (status_flags & GODMODE) || HAS_TRAIT(src, TRAIT_NOCLONELOSS)) )
 		return FALSE
 	cloneloss = CLAMP((cloneloss + (amount * CONFIG_GET(number/damage_multiplier))), 0, maxHealth * 2)
 	if(updating_health)
 		updatehealth()
 	return amount
 
 /mob/living/proc/setCloneLoss(amount, updating_health = TRUE, forced = FALSE)
-	if(!forced && (status_flags & GODMODE))
+	if(!forced && (status_flags & GODMODE) || HAS_TRAIT(src, TRAIT_NOCLONELOSS))
```
Comment (spookydonut) [inline code/modules/mob/living/damage_procs.dm]:
```suggestion
	if(!forced && ( (status_flags & GODMODE) || HAS_TRAIT(src, TRAIT_NOCLONELOSS)) )
```

**Pull request merged**